hydromt/data_adapter/data_adapter.py

```diff
@@ -125,6 +125,8 @@ def __init__(
         driver_kwargs={},
         name="",  # optional for now
         catalog_name="",  # optional for now
+        provider="UNSPECIFIED",
+        data_version="UNSPECIFIED",
```

Comment:
Because `None` cannot be a key in a dictionary, and in the data catalog we use those arguments as keys. We could have the empty string like the two above, but I think that is much more confusing when we write these things out back to catalogs. at least UNSPECIFIED is clear and also easy to search for in logs e.g. I will add the type hints, that's a good remark

Reply:
actually scratch that, I see now I use `None` as an argument in other places. hang on let me rethink my answer